ozaria/site/components/char_customization/PageCharCustomization/index.vue

```diff
@@ -0,0 +1,318 @@
+        const selectedThang = ozariaHeroes[this.selectedHero].thang
+        selectedThang.getLankOptions = () => {
+          const options = { colorConfig: {} }
+          const playerTints = [ this.tintBySlug('hair')[this.tintIndexSelection.hair], this.tintBySlug('skin')[this.tintIndexSelection.skin] ]
+          playerTints.forEach(tint => {
+            options.colorConfig = _.merge(options.colorConfig, tint)
+          })
+          return options
+        }
+        return selectedThang
+      }
+    },
+
+    methods: {
+      ...mapActions('tints', ['fetchTints']),
+
+      setInitialData () {
+        const ozariaHeroConfig = me.get('ozariaHeroConfig') || {}
+        this.characterName = ozariaHeroConfig.playerHeroName || ''
+
+        if (ozariaHeroConfig.thangType) {
+          for (const key of Object.keys(ozariaHeroes)) {
+            if (ozariaHeroConfig.thangType === ozariaHeroes[key].original) {
+              this.selectedHero = key
+            }
+          }
+        } else {
+          this.selectedHero = Object.keys(ozariaHeroes)[Math.floor(Math.random() * Object.keys(ozariaHeroes).length)]
+        }
+
+        for (const { slug, colorGroups } of ozariaHeroConfig.tints) {
+          const allowedTints = this.tintBySlug(slug)
+          for (let i = 0; i < allowedTints.length; i++) {
+            if (_.isEqual(allowedTints[i], colorGroups)) {
+              Vue.set(this.tintIndexSelection, slug, i)
+            }
+          }
+
+          if (this.tintIndexSelection[slug] === -1) {
+            Vue.set(this.tintIndexSelection, slug, Math.floor(Math.random() * allowedTints.length))
+          }
+        }
+      },
+
+      selectBodyType (hero) {
+        this.selectedHero = hero
+      },
+
+      tintBySlug (slug) {
+        return ((this.characterCustomizationTints.find(t => t.slug === slug) || {}).allowedTints || [])
+      },
+
+      onClickSwatch (slug, i) {
+        Vue.set(this.tintIndexSelection, slug, i)
+      },
+
+      handleSubmit () {
+        const valid = this.$refs['name-form'].reportValidity()
+        const name = this.characterName.trim()
+        if (!valid) {
+          return
+        }
+        if (name === '') {
+          return noty({ text:"Invalid Name", layout: 'topCenter', type: 'error' })
+        }
+
+        const ozariaConfig = me.get('ozariaHeroConfig') || {}
+        ozariaConfig.playerHeroName = name
+
+        ozariaConfig.tints = [
+          {
+            slug: 'hair',
+            colorGroups: this.tintBySlug('hair')[this.tintIndexSelection.hair]
+          },
+          {
+            slug: 'skin',
+            colorGroups: this.tintBySlug('skin')[this.tintIndexSelection.skin]
+          }
+        ]
+
+        ozariaConfig.thangType = this.ozariaHeroes[this.selectedHero].original
+
+        me.set('ozariaHeroConfig', ozariaConfig)
+        // TODO button doesn't have any behavior yet.
+        me.save(null, {
+          success: () => {
+            this.$emit('completed')
```

Comment:
It might not actually be needed to emit this event since its not going to be part of the intro level and nothing is listening to this event as far as I know.

Reply:
Will remove and just leave the comment.